Aztec/Classes/TextKit/TextView.swift

```diff
@@ -724,6 +767,56 @@ open class TextView: UITextView {
     }
 
 
+    /// Removes the List Attributes from a collection of attributes, whenever:
+    ///
+    ///     A. The caret is at the very end of the document (+ there is no selected text)
+    ///     B. There's a list!
+    ///     C. The previous character is a '\n'
+    ///
+    /// - Parameter attributes: Typing Attributes.
+    ///
+    /// - Returns: Updated Typing Attributes.
+    ///
+    private func ensureRemovalOfListAttribute(from attributes: [String: Any]) -> [String: Any] {
+        guard selectedRange.location == storage.length && selectedRange.length == 0 else {
+            return attributes
+        }
+
+        guard TextListFormatter.listsOfAnyKindPresent(in: attributes) else {
+            return attributes
+        }
+
+        let previousRange = NSRange(location: selectedRange.location - 1, length: 1)
+        let previousString = storage.safeSubstring(at: previousRange) ?? String(.newline)
+        guard previousString == String(.newline) else {
+            return attributes
+        }
+
+        var updatedAttributes = attributes
+        updatedAttributes[NSParagraphStyleAttributeName] = ParagraphStyle.default
+
+        return updatedAttributes
+    }
+
+
+    /// Inserts an empty line, whenever we're at the end of the document, and there's no selected text.
+    ///
+    private func ensureInsertionOfNewlineOnEmptyDocuments() {
+        guard selectedRange.location == storage.length && selectedRange.length == 0 else {
+            return
+        }
+
+        // Note: We *really* need to use super, so that we prevent recursive loops.
```

Comment:
For the record:

Removed the `super.typingAttributes` (it was probably crashing due to testing code!), but left the `super.insertText`, since, without that one, we do get a loop.

Re: extra docs, as discussed over Slack, we've got a unit test that would trigger the loop, so we're cool.

Thanks a lot Diego!